src/web_scraper/web_scraper.py

```diff
@@ -0,0 +1,125 @@
+"""
+
+Description: this program will eventually be responsible for extracting RSS senate/house bill metadata and DOCX full bill, and then uploading to supabase table
+
+
+"""
+
+from docx import Document
+from supabase import create_client, Client
+from datetime import datetime
+import feedparser
+import os
+import re
+
+senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
+house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
+
+sb_api_url = "https://vsumrxhpkzegrktbtcui.supabase.co" #os.environ["API_URL"] 
+sb_api_key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZzdW1yeGhwa3plZ3JrdGJ0Y3VpIiwicm9sZSI6ImFub24iLCJpYXQiOjE2ODQ3OTU2NzUsImV4cCI6MjAwMDM3MTY3NX0.9lafalZT9FJW1D8DAuIMrsRX0Gs6204nV8ETfGslrqI" #os.environ["API_KEY"] 
+
+def extract_and_upload_senate_bill_metadata():
+    sen_rss = feedparser.parse(senate_rss_feed)
+    supabase: Client = create_client(sb_api_url, sb_api_key)
+    for bill in sen_rss.entries:
+        #print("checking bill ", bill.title)
+        if bill is None: continue
+
+        bill_no_regex = '(Senate Bill \d*|Senate Resolution \d*)'
+        bill_no = re.findall(bill_no_regex, bill.title)[0].split(' ')[-1]
+        
+        printer_no_regex="Printer's Number \d*"
+        printer_no = re.findall(printer_no_regex, bill.title)[0].split(' ')[-1]
+
+        #OLD CODE that doesn't work
+        #title_elements = bill.title.split(' ') #grabbing bill and printer numbers from the title (not explicitly given in RSS feed)
+        #bill_no = "".join(char for char in title_elements[2] if char.isdigit()) #ensuring that only the actual digits of the bill number are extracted
+        #printer_no = "".join(char for char in title_elements[5] if char.isdigit()) #ensuring that only the actual digits of the printer number are extracted
+
+        #change if bill["parss_enacted"] == "YES" and title_elements[1] == "Bill": #make sure the bill was enacted and it's a bill instead of a resolution        
```

Comment:
For this old code that doesn't work, let's just delete it rather than keeping it in the script. If we need to, we can review it in version control.